Pipfile

```diff
@@ -8,6 +8,7 @@ clip = {git = "https://github.com/openai/CLIP.git"}
 torch = "==1.9.0+cpu"
 torchvision = "==0.10.0+cpu"
 pillow = ">=8.3.2"
+requests = "~=2.26"
```

Comment:
Struggling to get this to resolve without errors.


PS -- wondering what direction you want to take this project in the future.    On my home machines it's now the main way I've been organizing my pictures; and I've hacked on a lot of features that might-or-might-not be out of scope for your vision for this project - including:

- De-dup ---  Embeddings are cool - they can even find duplicate thumbnails of different sizes.  I added a script that finds pictures with identical (or near identical) embeddings, and flag them as a duplicate in a column I added to SQLite.
- Support for explicit captions/tags/picking up words from folders --- for some images I already had them tagged with words that CLIP couldn't figure out.  For example, folders of country names I've visited.   I added a column for the CLIP embedding of the words I've manually tagged; and optionally also search that column.   Without it, a search for 'germany' only returns about 1/10 of my pictures from germany.
- Face embeddings -- so I can search for a specific person.  I'm using https://github.com/serengil/deepface to add a column for faces.  Currently I can't input the name of a face - but using the "similar to an existing image" I can find them.   Combining this with the "explicit tagging" feature above I'll soon be able to search for people by name (as long as I have tagged them at least once)

No idea if you consider those in or our of scope for the main project.

Reply:
@ramayer, thanks for the update. I'll take a look at it.

About the direction:
1. I would love to keep the rclip as a small and simple CLI tool, but considering building a more advanced app on top of it.
2. I like the de-dup idea and think that it can be a part of the rclip CLI tool. But, I am not 100% sure about this one yet, so instead of filing a GHI, I created a Discussion here: https://github.com/yurijmikhalevich/rclip/discussions/23.
3. This is a must-have feature for a practical photo search tool, but implementing it will fundamentally change how `rclip` works. Also, if we will implement this in `rclip` it will start output pictures that don't feature anything German in them (if though they were taken in Germany; for example, a picture of a generic bed made in the hotel). I feel like this feature can be a part of another tool that will incorporate `rclip`. Or I should change how I think about `rclip` :sweat_smile:  Let me take to some time to consider this. I also think that we may consider extracting manual/existing tags from the images themselves.
4. Face embeddings feature sounds great, and my general attitude to it is more open compared to the one I have towards indexing the folder names.

Thank you for sharing all these. All good ideas that I need to consider.

Reply:
@ramayer, `pipenv install "requests~=2.26"` worked fine for me. But, you shouldn't forget to execute `pipenv shell` before doing anything with `pipenv`. I mention it here, in the Contributing section: https://github.com/yurijmikhalevich/rclip#contributing. Executing `pipenv shell` sets env variable from `.env` which is required to install the PyTorch with `pipenv` from the official PyTorch "repository."